common/workunit/workunit.cpp

```diff
@@ -2446,7 +2426,11 @@ class CDaliWorkUnitFactory : public CWorkUnitFactory, implements IDaliClientShut
     virtual void deleteRepository(bool recreate)
     {
         Owned<IRemoteConnection> conn = sdsManager->connect("/WorkUnits", session, RTM_LOCK_WRITE, SDS_LOCK_TIMEOUT);
-        conn->close(true);
+        if (conn)
+            conn->close(true);
+        conn.setown(sdsManager->connect("/GraphProgress", session, RTM_LOCK_WRITE, SDS_LOCK_TIMEOUT));
+        if (conn)
+            conn->close(true);
     }
     virtual void createRepository()
```

Comment:
None of my unit testing threw up any places that assumed workunits existed, and as far as I can tell from a quick search of the code all usage either fails cleanly when it does not exist (i.e. when trying to read a workunit, where by definition if the root is not there neither is the workunit) or uses QUERY_CREATE mode. I think can be left as it is. I don't expect a lot of places beyond my unit test program to be calling this function anyway!